tiktok.cs

```diff
@@ -5,394 +5,105 @@ namespace First_project
-                    {
-
-                        if (board[xGuess, yGuess] != 0)
-                        {
-                            Console.WriteLine("Already used this square! Pick another.");
-                            XTurn(out xGuess, out yGuess);
-                        }
-                        else
-                        {
-                            board[xGuess, yGuess] = 1;
-                            DrawBoard(board);
-                            if (CheckVictoryX(board, xGuess, yGuess) == true)
-                                {
-                                goto End;
-                            }
-
-
-                        }
-                    }
-                    turnCounter += 1;
-                    //YTurn
-                    YTurn(out xGuess, out yGuess);
-
-                    if (xGuess > 3)
-                    {
-                        Console.WriteLine("That's not a square! Try again.");
-                        YTurn(out xGuess, out yGuess);
-                    }
-                    if (xGuess < 1)
-                    {
-                        Console.WriteLine("That's not a square! Try again.");
-                        YTurn(out xGuess, out yGuess);
-                    }
-                    else
+                int turnCounter = 0;
+                bool invalidTurn = false;
+                bool draw = true;
+                DrawBoard(board);
+                while (turnCounter <= 8)
+                {
+                    int playerToken = turnCounter % 2 + 1;
+
+                    do{
+                      Turn(out xGuess, out yGuess, playerToken);
+                      if (xGuess > 2 || xGuess < 0 || yGuess > 2 || yGuess < 0) 
+                      {
+                         Console.WriteLine("That's not a square! Try again.");
+                         invalidTurn = true;
+                      }
+                      else if(board[xGuess, yGuess] != 0){
+                        Console.WriteLine("Already used this square! Pick another.");
+                        invalidTurn = true;
+                      }
+                      else{
+                        invalidTurn = false;
+                      }
+                    }while(invalidTurn);
+
+                    board[xGuess, yGuess] = playerToken;
+                    DrawBoard(board);
+
+                    if (CheckVictory(board, playerToken))
                     {
-
-                        if (board[xGuess, yGuess] != 0)
-                        {
-                            Console.WriteLine("Already used this square! Pick another.");
-                            YTurn(out xGuess, out yGuess);
-                        }
-                        else
-                        {
-                            board[xGuess, yGuess] = 2;
-                            DrawBoard(board);
-                            if (CheckVictoryY(board, xGuess, yGuess) == true)
-                            { goto End; }

```

Comment:
Never. Ever. Use goto statements. Always use loops, functions or conditional checks instead. This is the topic of one of the most famous papers in computer science, called [Goto Considered Harmful](https://homepages.cwi.nl/~storm/teaching/reader/Dijkstra68.pdf)